config/policy.go

```diff
@@ -0,0 +1,248 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"golang.ngrok.com/ngrok/internal/pb"
+)
+
+type policy struct {
+	Inbound  inboundRules  `json:"inbound,omitempty"`
+	Outbound outboundRules `json:"outbound,omitempty"`
+}
+
+type inboundRules []policyRule
+type outboundRules []policyRule
+
+type policyRule struct {
+	Name        string   `json:"name,omitempty"`
+	Expressions []string `json:"expressions,omitempty"`
+	Actions     []action `json:"actions"`
+}
+type action struct {
+	Type   string          `json:"type"`
+	Config json.RawMessage `json:"config,omitempty"`
+}
+
+type PolicyActionOption = option[*action]
+type PolicyRuleOption = option[*policyRule]
+type PolicyRuleSetOption = option[*[]policyRule]
+type PolicyOption = option[*policy]
+
+// Supports conversion to a json string
+type JsonConvertible interface {
+	ToJSON() string
+}
+
+func (p *policy) ToJSON() string {
+	return toJSON(p)
+}
+
+func (p policyRule) ToJSON() string {
+	return toJSON(p)
+}
+
+func (p action) ToJSON() string {
+	return toJSON(p)
+}
+
+func toJSON(o any) string {
+	bytes, err := json.Marshal(o)
+
+	if err != nil {
+		panic(fmt.Sprintf("failed to convert to json with error: %s", err.Error()))
+	}
+
+	return string(bytes)
+}
+
+// WithPolicyConfig configures this edge with the provided policy configuration
+// passed as a json string and overwrites any prevously-set traffic policy
+// https://ngrok.com/docs/http/traffic-policy
```

Comment:
This mentions `traffic-policy`, wanted to double check that that name is current (I just haven't heard `traffic` before here, and that docs link isn't live).

Reply:
That is indeed the latest on what we are calling the module itself "Traffic Policy" but all the configuration structures just refer to it as "policy"